Implement e2e accessibility tests for pages in DSpace User Inteface

FrancescoMolinaro opened the pull request:
## References
Resolve issue: https://github.com/DSpace/dspace-angular/issues/2780

## Description
Implemented automated accessibility tests for pages in User Interface and for pages accessible also to anonymous users which missed tests.

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You do not need to complete this checklist prior creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [ ] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [ ] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [ ] My PR **passes [ESLint](https://eslint.org/)** validation using `yarn lint`
- [ ] My PR **doesn't introduce circular dependencies** (verified via `yarn check-circ-deps`)
- [ ] My PR **includes [TypeDoc](https://typedoc.org/) comments** for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR **passes all specs/tests and includes new/updated specs or tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] My PR **aligns with [Accessibility guidelines](https://wiki.lyrasis.org/display/DSDOC8x/Accessibility)** if it makes changes to the user interface.
- [ ] My PR **uses i18n (internationalization) keys** instead of hardcoded English text, to allow for translations.
- [ ] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

tdonohue reviewed (commented):
@FrancescoMolinaro : Apologies for the delay in reviewing this.  This looks **great**!  I've run the tests locally to be sure they work, and almost all of them work perfectly.  However, I'm seeing minor accessibility failures from `processes-overview.cy.ts` and `admin-notifications-publication-claim.cy.ts`... But, it's possible it's my local test data causing issues.

Could you perhaps rebase this on the latest `main` in order to trigger the e2e tests to run again?  If a fresh run is successful, we can merge this immediately.  Thanks again.

FrancescoMolinaro commented:
Hi @tdonohue , no worries at all and thanks for checking.
I have rebased the branch with main  and noticed there were still a couple of issue:

1) **item-edit** would often go in timeout, I have added an interceptor to mitigate this issue and at least locally I can't reproduce the timeout anymore.

2) **edit-group** would sometime throw an accessibility error, landmark-unique, because of the multiple pagination components in the page. I have added an aria-label composed by a brief descriptive text and the ID of the pagination options, in this way each and every component should have now a unique descriptive text (role + aria-label).

A part from these two fixes I haven't experienced issues with **processes-overview** or **admin-notifications-publication-claim** and also on the latest run I can see they are passing successfully, it could actually depends on your local data.

tdonohue reviewed (approved):
👍 Thanks @FrancescoMolinaro ! This looks great now.  I'm going to merge immediately, and see if we can backport it (minimally to 8.x).  A backport to 7.x might need to be done manually (because not all the features tested in this PR existed in 7.x), but would also be worthwhile

tdonohue commented:
@FrancescoMolinaro : Could you see if this is possible to port to `dspace-7_x`?  As I expected, the port for that failed.  I think it is likely because some of the features/changes tested here only exist in 8.x.   So, I'm hopeful that a `dspace-7_x` version of this PR should be smaller.

tdonohue commented:
Ported to 7.x in #3405
